.github/workflows/gen.yml

```diff
@@ -0,0 +1,73 @@
+name: "Generate"
+on:
+  push:
+    paths-ignore:
+      - 'acceptancetests/**'
+      - 'doc/**'
+      - 'snap/**'
+      - 'testcharms/**'
+      - 'testing/**'
+      - 'tests/**'
+  pull_request:
+    types: [opened, synchronize, reopened, ready_for_review]
+    paths-ignore:
+      - 'acceptancetests/**'
+      - 'doc/**'
+      - 'snap/**'
+      - 'testcharms/**'
+      - 'testing/**'
+      - 'tests/**'
+  workflow_dispatch:
+
+permissions:
+  contents: read
+
+jobs:
+  Generate:
+    name: Generate
+    runs-on: ubuntu-latest
+    if: github.event.pull_request.draft == false
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Find required go version
+        id: go-version
+        run: |
+          set -euxo pipefail
+          echo "::set-output name=version::$(grep '^go ' go.mod | awk '{print $2}')"
+
+      - name: Set up Go
+        uses: actions/setup-go@v3
+        with:
+          go-version: ${{ steps.go-version.outputs.version }}
+        id: go
+
+      - name: go get dependencies
+        run: go get ./...
+
+      - name: Delete all mocks
```

Comment:
Is this so that mocks that are missing a go generate directive get flagged?

Reply:
Yep that's right. Pretty sure we had some that should no longer be there